python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -280,19 +280,27 @@ def __exit__(
                 exception_value, (DagsterUserCodeUnreachableError, DagsterCodeLocationLoadError)
             ):
                 try:
-                    raise Exception(
+                    raise DagsterUserCodeUnreachableError(
                         "Unable to reach the code server. Automation condition evaluation will resume once the code server is available."
                     ) from exception_value
                 except:
-                    error_data = DaemonErrorCapture.on_exception(sys.exc_info())
+                    error_data = DaemonErrorCapture.on_exception(
```

Comment:
`DaemonErrorCapture.on_exception` reads a little weird  as the "always call for exceptions" fn . Its too bad all the calls arent in deaemon classes otherwise i would say just reshuffle to `self._exception` defined on the daemon base class. Maybe could still do that but pass the callback instead of logger when the code is outside the daemon class? You can call sys.exc_info in the fn so would just need to pass log msg at callsites. 

otherwise moving to something like `DagsterDaemon.process_exception` might feel better?

Reply:
moved to process_exception